frozen_mutation.hh

```diff
@@ -102,22 +102,22 @@ public:
         r.append_cell(id, collection_mutation(*_schema.static_column_at(id).type, std::move(collection)));
     }
 
-    virtual void accept_row_tombstone(range_tombstone rt) override {
+    virtual stop_iteration accept_row_tombstone(range_tombstone rt) override {
         flush_rows_and_tombstones(rt.position());
         _rt_gen.consume(std::move(rt));
-        // FIXME: propagate stop_consuming to caller
+        return _stop_consuming;
```

Comment:
In one place you check the result of flush_rows_and_tombstones(), here you check _stop_consuming. Perhaps we should stick to one method.